static/app/views/settings/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -284,7 +284,8 @@ export class SentryAppExternalForm extends Component<Props, State> {
     const choiceArray = await Promise.all(
       impactedFields.map(field => {
         // reset all impacted fields first
-        this.model.setValue(field.name || '', '', {quiet: true});
+        const defaultValue = this.getDefaultFieldValue(field);
```

Comment:
Just trying to understand this better. Where does skip_on_load come to play?

Reply:
Er so it's a little complicated. I was wrong earlier today. How `skip_load_on_open` works is that, we will not make the 3p call to get the field options until the [field has been typed in](https://docs.sentry.io/organization/integrations/integration-platform/ui-components/formfield/#select). By setting a default value to this field, we will automatically make a call to the 3p for external values since there is some input/'typing' there. I assume that's why this line was originally resetting the field to empty string, since any value == we make a 3p call. Though, I think this new side effect is okay, since users will expect having a default value in a form means we've already done some work to query that value (as seen with bug 2).

Reply:
Got it. So one more thing I would test is editing a field that is not empty to make it empty 🤔  save and would it stay empty. I wonder if default value overrides it.

Reply:
is that possible ? A select field has a list of choices that the user can pick so I don't think we can make it empty ? 🤔

Reply:
Though, I guess if the field is optional and the person doesn't want to have any input in the field. Then yeah that's a valid a test case... seeing if we can delete the input.

Reply:
oof yeah we can't currently in this PR💀 I'll get to fixin that

Reply:
oh wait nevermind, you can have an empty value/reset the field by clicking the x in the far right of the select field. You can't do it via typing though.